tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs

```diff
@@ -42,6 +42,7 @@ fn test_create_ak_rsa_rsa() {
 }
 
 #[test]
+#[should_panic]
```

Comment:
I don't understand this panic. Should you not test that the correct error is generated instead?

Reply:
I guess Ionut did so because I suggested it in a previous comment. Sorry 🙇‍♂️! Maybe indeed `assert!(matches!(...` would be a better option 🤔

Reply:
I mean for me ```#[should_panic]``` is used to test that the library panics under certain conditions. And we try very hard to not panic any where in the library so that should hopefully not happen.

Things that one can do to test is:
1. If the error in it self is not that import but rather that an error is returned at all then you can:
```rust
let result = ak::foo();
assert!(result.is_err(), "Function fo should have returned an error.");
```
This is not done any where in our code currently because the returned errors are a part of the API.

2. The specific error is important:
```rust
let result = ak::foo();
if let Err(actual_error) = result {
    assert_eq!(
        Error::WrapperError(WrapperErrorKind::InvalidParam),
        actual_error,
        "Foo did not produce the expected error."
    );
} else {
    panic!("Foo is expected to return an error");
}
```
This is done extensively when testing the TSS return code errors.

Reply:
Thanks for the pushback, should've thought this through 😅 

I'll tweak it and push back up.

Reply:
Right, only took about 3 months, hope it works!